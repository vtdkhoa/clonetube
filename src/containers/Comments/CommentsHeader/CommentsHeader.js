import React from 'react'
import { Button, Icon } from 'semantic-ui-react'
import PropTypes from 'prop-types'
import './CommentsHeader.scss'

const CommentsHeader = props => {
  const amountComments = Number(props.amountComments).toLocaleString() || ''

  return (
    <div className="comments-header">
      <h4>{amountComments} Comments</h4>
      <Button basic compact icon labelPosition="left">
        <Icon name="align left"/>
        Sort by
      </Button>
    </div>
  )
}

CommentsHeader.propTypes = {
  amountComments: PropTypes.any
}

export default CommentsHeader