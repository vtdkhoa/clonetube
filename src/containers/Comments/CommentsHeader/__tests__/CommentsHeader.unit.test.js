import React from 'react'
import CommentsHeader from '../CommentsHeader'
import { shallow } from 'enzyme'

describe('CommentsHeader', () => {
  test('renders with amountComments = null', () => {
    const wrapper = shallow(<CommentsHeader/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with amountComments = 1000000', () => {
    const wrapper = shallow(
      <CommentsHeader amountComments={1000000}/>
    )
    expect(wrapper).toMatchSnapshot()
  })
})