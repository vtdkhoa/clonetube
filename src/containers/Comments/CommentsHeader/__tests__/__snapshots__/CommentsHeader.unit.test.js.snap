// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsHeader renders with amountComments = 1000000 1`] = `
<div
  className="comments-header"
>
  <h4>
    1.000.000
     Comments
  </h4>
  <Button
    as="button"
    basic={true}
    compact={true}
    icon={true}
    labelPosition="left"
  >
    <Icon
      as="i"
      name="align left"
    />
    Sort by
  </Button>
</div>
`;

exports[`CommentsHeader renders with amountComments = null 1`] = `
<div
  className="comments-header"
>
  <h4>
    NaN
     Comments
  </h4>
  <Button
    as="button"
    basic={true}
    compact={true}
    icon={true}
    labelPosition="left"
  >
    <Icon
      as="i"
      name="align left"
    />
    Sort by
  </Button>
</div>
`;
